### DISCUSSION TITLE
Should vcpkg_configure|install_meson be moved into vcpkg-tool-meson

### DISCUSSION THREAD

**Discussion by Neumann-A** [microsoft/vcpkg#28808] [Polls]
So the question is if `vcpkg_configure|install_meson` and other meson related scripts should simply be moved into `vcpkg-tool-meson` and then be loaded/overwritten via a `vcpkg-port-config.cmake`. All ports dependening on meson should already depend on `vcpkg-tool-meson` and moving the script seems to be the correct choice. 
The idea is to keep a unmainted copy in `scripts/cmake` but have the same function be defined in `vcpkg-tool-meson`. Since the general scripts are loaded before the port configs versioning of these scripts could be achieved without braking older versions.

**Comment by Osyotr** [comment]
If there are no downsides, why not?

**Reply by Neumann-A** [reply]
the downside is using a undocumented cmake debugging feature. But vcpkg.cmake also uses it so .....

**Comment by JackBoosY** [comment]
I guess we will move all the configure / build / install functions into ports sooner or later.

**Reply by JackBoosY** [reply]
On the other hand, the changes in these functions are smaller than other ports. I wish we could separate the *real* port to another repo.

**Comment by dg0yt** [comment]
Shouldm't the script functions be in `vcpkg-meson` script port, in contrast to the `vcpkg-tool-meson` binary port?

**Reply by Neumann-A** [reply]
I don't think that detail matters but the script probably needs to be more or less tightly coupled with the meson version. So it is highly likely that a meson update requires script changes which might be incompatible with older/newer versions of mesons.

**Reply by dg0yt** [reply]
I don't want to overrate it, but a) there is consistency (vcpkg-cmake, vcpkg-gn)  and b) users might want or need to overlay a tool port separately from a script port.
